recommender/engine/rl_agent/models/actor_model.py

```diff
@@ -1,37 +1,81 @@
+# pylint: disable=missing-module-docstring, invalid-name, too-many-arguments, no-member
+
+from typing import Tuple
+from itertools import chain
+
 import torch
+from torch import nn
+
+from recommender.engine.rl_agent.models.history_model import HistoryModel
+from recommender.engine.rl_agent.models.search_phrase_model import SearchPhraseModel
 
 
-class ActorModel(torch.nn.Module):
-    def __init__(self, K: int, SE: int, history_embedder: torch.nn.Module):
+class ActorModel(nn.Module):
+    """Actor neural network representing a deterministic policy used by RLAgent"""
+
+    def __init__(
+        self,
+        K: int,
+        SE: int,
+        UE: int,
+        SPE: int,
+        search_phrase_embedder: SearchPhraseModel,
+        history_embedder: HistoryModel,
+        layer_sizes: Tuple[int] = (256, 512, 256),
+    ):
+        """
+        Args:
+            K: number of services to recommend
+            SE: service embedding dimension and filters embedding dimension
+            UE: user embedding dimension
+            SPE search phrase embedding dimension
+            history_embedder: pytorch module implementing history embedding
+            layer_sizes: list containing number of neurons in each hidden layer
+        """
+
         super().__init__()
         self.K = K
         self.SE = SE
         self.history_embedder = history_embedder
```

Comment:
History embedder pytorch model and Searchphrase Embedder pytorch model should be automatically loaded from the database if provided parameters are None. But this can be done in another issue - after refactoring.